### ISSUE TITLE
Enable readonly-access to Index from multiple threads.

### ISSUE THREAD

**Issue by xivk** [itinero/reminiscence#6]
Related to:

https://github.com/itinero/routing/issues/55

**Comment by juliusfriedman** [comment]
Is this related to https://github.com/itinero/routing/issues/256 as well?

**Comment by juliusfriedman** [comment]
@xivk, https://github.com/itinero/reminiscence/pull/23 highlights a test which fails when accessing index in parallel, I tried to look into it but couldn't get far as of yet. I made my comments in the PR.

I am assuming this issue is related to that same thing or is this a new issue?

**Comment by juliusfriedman** [comment]
Also if the buffer is used couldn't there be a caching mechanism which checks to see if the value is already in the buffer and just read it from there rather than the stream? I am not quite sure how to implement this ... It might already be implemented to some degree as that unit test shows, after first enumeration the parallel tests don't exhibit the issue...
